unfetter-discover/unfetter-ui
# framework switch

supports unfetter-discover/unfetter#855

## Problem
To allow new attack patterns to live side by side in unfetter w/ its default attack patterns

## Accomplished
* Fix the threat dashboard to show only the user preferred attack patterns
* Move the kill chain drop down to the settings page, because the profile page is more widely accessible across users

## To Test
* verify travis run
* pull this code
* visit your settings page in uac mode
* select a framework preference
* look in mongo `db.getCollection('user').find({}, { 'preferences.killchain': 1 })` and verify your entry
* visit the threat dashboard, and ensure the creation and existing entries work
### Additionally 
* uncomment the ntctf in `docker-compose.development.yml`
* restart the stack
* visit the threat dashboard
* create a report, notice the attack patterns are ntctf
* create a workproduct notice the threat dashboard shows only ntctf phases for ntctf reports and nothing for non ntctf reports
